pennylane/devices/experimental/device_api.py

```diff
@@ -341,12 +341,12 @@ def supports_derivatives(
 
         """
         if execution_config is None:
-            return self.compute_derivatives != Device.compute_derivatives
+            return type(self).compute_derivatives != Device.compute_derivatives
```

Comment:
was this change necessary/did some test fail? I know it's silly but are we intentionally avoiding cases like:
```python
class SomeUniqueDevice(Device):
    def computes_derivatives(self):
        something()

dev = SomeUniqueDevice()
dev.computes_derivatives = Device.computes_derivates
```

Reply:
So when you instantiate the device, the method becomes a new thing. In order to check "was this overridden" we need to check what it is on the type, not the instance.